src/Motions/Document.ts

```diff
@@ -20,6 +20,19 @@ export class MotionDocument extends Motion {
         return MotionDocument.toLine({n: args.n === undefined ? +Infinity : args.n});
     }
 
+    static toLinePercent(args: {n: number}): Motion {
+        const activeTextEditor = window.activeTextEditor;
+        let line
+        if (! activeTextEditor ) {
+            line = 0
+        }else{
+            const document = activeTextEditor.document;
+            line = Math.round(document.lineCount * args.n / 100)
+            line = Math.min(document.lineCount, line)
+        }
```

Comment:
The document may have changed when the Motion is applied(by calling `apply` method), so we should save the percentage(make sure it's in range 0-1) and calculate the line number in `apply`.

Reply:
Besides, the `{N}%` motion also set the cursor to the first non-blank character.
You can refer to `MotionLine.firstNonBlank` for similar implementation.